### PR TITLE
%%ml analyze can take a tarball 

### PR DESCRIPTION
**Pull request by brandondutra**
* add support for taking a tarball as the source
* will download the tarball if on gcs
* untar's it, and analyze uses the analyze.py file

todo

* support other file formats like .zip, .tar.bz, .tar.Z
* do this to transform and training too
* once we figure out the documentation story, we need to document the expected format of the tarball (analyze.py, transform.py, setup.py, trainer/task.py)

### CODE REVIEW

File: google/datalab/contrib/mltoolbox/commands/_ml.py [hunk 1]
```diff
@@ -129,7 +130,8 @@ def ml(line, cell=None):
                               help='path of output directory.')
   analyze_parser.add_argument('--cloud', action='store_true', default=False,
                               help='whether to run analysis in cloud or local.')
-
+  analyze_parser.add_argument('--source_code', required=False,
```
Comment (qimingj) [inline google/datalab/contrib/mltoolbox/commands/_ml.py]: --package?

File: google/datalab/contrib/mltoolbox/commands/_ml.py [hunk 2]
```diff
@@ -129,7 +130,8 @@ def ml(line, cell=None):
...
+                              help='A local or GCS tarball path to use as the source.')
```
Comment (qimingj) [inline google/datalab/contrib/mltoolbox/commands/_ml.py]: Also add that if missing, the default mltoolbox package is used?

File: google/datalab/contrib/mltoolbox/commands/_ml.py (hunk 1 above)
Comment (brandondutra) [inline google/datalab/contrib/mltoolbox/commands/_ml.py]: done

File: google/datalab/contrib/mltoolbox/commands/_ml.py (hunk 2 above)
Comment (brandondutra) [inline google/datalab/contrib/mltoolbox/commands/_ml.py]: done

**Comment by brandondutra** [comment]
ptal

**Pull request merged**